Marlin/src/gcode/control/M3-M5.cpp

```diff
@@ -65,76 +65,56 @@
  *
  *  PWM duty cycle goes from 0 (off) to 255 (always on).
  */
+
 void GcodeSuite::M3_M4(const bool is_M4) {
+  planner.synchronize();   // Wait for previous movement commands (G0/G1/G2/G3) to complete before changing power
+  if (parser.seen('I')) {
+    cutter.cutter_mode = is_M4 ? CUTTER_MODE_DYNAMIC : CUTTER_MODE_CONTINUOUS;
+    cutter.set_enabled(true);
+  }
+
   auto get_s_power = [] {
-    if (parser.seenval('S')) {
-      const float spwr = parser.value_float();
+    if (parser.seen('S')) {
+      cutter.unitPower = parser.value_ushort();
+      // PWM implied and ranges from S0 to S180 for a positional servo. Typical use would be a pen up/down function.
       #if ENABLED(SPINDLE_SERVO)
-        cutter.unitPower = spwr;
+        cutter.power = cutter.unitPower;
       #else
-        cutter.unitPower = TERN(SPINDLE_LASER_PWM,
-                              cutter.power_to_range(cutter_power_t(round(spwr))),
-                              spwr > 0 ? 255 : 0);
+        if (cutter.cutter_mode == CUTTER_MODE_STANDARD) // PWM not implied, power converted to OCR from unit definition and min/max or on/off if not PWM.
+          cutter.power = TERN(SPINDLE_LASER_PWM, cutter.power_to_range(cutter_power_t(cutter.unitPower)), cutter.unitPower > 0 ? 255 : 0);
       #endif
+      cutter.menuPower = cutter.unitPower;
     }
-    else
+    else if (cutter.cutter_mode == CUTTER_MODE_STANDARD)
       cutter.unitPower = cutter.cpwr_to_upwr(SPEED_POWER_STARTUP);
+
     return cutter.unitPower;
   };
 
-  #if ENABLED(LASER_POWER_INLINE)
-    if (parser.seen('I') == DISABLED(LASER_POWER_INLINE_INVERT)) {
-      // Laser power in inline mode
-      cutter.inline_direction(is_M4); // Should always be unused
-      #if ENABLED(SPINDLE_LASER_PWM)
-        if (parser.seen('O')) {
-          cutter.unitPower = cutter.power_to_range(parser.value_byte(), 0);
-          cutter.inline_ocr_power(cutter.unitPower); // The OCR is a value from 0 to 255 (uint8_t)
-        }
-        else
-          cutter.inline_power(cutter.upower_to_ocr(get_s_power()));
-      #else
-        cutter.set_inline_enabled(true);
-      #endif
-      return;
-    }
-    // Non-inline, standard case
-    cutter.inline_disable(); // Prevent future blocks re-setting the power
-  #endif
-
-  planner.synchronize();   // Wait for previous movement commands (G0/G0/G2/G3) to complete before changing power
-  cutter.set_reverse(is_M4);
```

Comment:
Are you maintaining proper consideration of the **reverse spindle** `M4` in removing this line?

Reply:
Good catch, that's an error.

Reply:
Updated